app/src/main/java/com/example/nvdovin/weatherapp/presentation/main/weather/forecast/adapter/ForecastRecyclerViewAdapter.java

```diff
@@ -0,0 +1,107 @@
+package com.example.nvdovin.weatherapp.presentation.main.weather.forecast.adapter;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import com.example.nvdovin.weatherapp.R;
+import com.example.nvdovin.weatherapp.data.model.WeatherData;
+import com.example.nvdovin.weatherapp.domain.model.CityForecast;
+import com.example.nvdovin.weatherapp.domain.utils.design.ImageUtils;
+import com.example.nvdovin.weatherapp.domain.utils.sharedpreferences.SharedPrefs;
+import com.example.nvdovin.weatherapp.domain.utils.temperature.TemperatureConvertor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class ForecastRecyclerViewAdapter extends RecyclerView.Adapter<ForecastRecyclerViewAdapter.CustomViewHolder> {
+    private static final int TRANSPARENCY_ALPHA = 160;
+    private final OnItemClickListener listener;
+    private ImageUtils imageUtils;
+    private List<CityForecast> cityForecastList;
+    private Context context;
+    private SharedPrefs sharedPrefs;
+
+    public ForecastRecyclerViewAdapter(List<CityForecast> cityForecastList, OnItemClickListener listener, Context context) {
+        this.context = context;
+        this.cityForecastList = cityForecastList;
+        this.listener = listener;
+        sharedPrefs = new SharedPrefs(context);
+        imageUtils = new ImageUtils(context);
+    }
+
+    @Override
+    public CustomViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(context).inflate(R.layout.forecast_recycler_row, parent, false);
+        return new CustomViewHolder(view);
```

Comment:
could you tell me please what *CustomViewHolder* means? Do you remember about meaningful naming?